### PR TITLE
[Documentation] Load Testing on DevNets and TestNets

### PR DESCRIPTION
**Pull request by okdas**
## Summary

Documents Load Testing on DevNet (and some TestNet?)

## Issue

- #136 
- #529

## Type of change

Select one or more:

- [ ] New feature, functionality or library
- [ ] Bug fix
- [ ] Code health or cleanup
- [x] Documentation
- [ ] Other (specify)

## Testing

**Documentation changes** (only if making doc changes)
- [x] `make docusaurus_start`; only needed if you make doc changes

**Local Testing** (only if making code changes)
- [ ] **Unit Tests**: `make go_develop_and_test`
- [ ] **LocalNet E2E Tests**: `make test_e2e`
- See [quickstart guide](https://dev.poktroll.com/developer_guide/quickstart) for instructions

**PR Testing** (only if making code changes)
- [ ] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.
    - **THIS IS VERY EXPENSIVE**, so only do it after all the reviews are complete.
    - Optionally run `make trigger_ci` if you want to re-trigger tests without any code changes
    - If tests fail, try re-running failed tests only using the GitHub UI as shown [here](https://github.com/pokt-network/poktroll/assets/1892194/607984e9-0615-4569-9452-4c730190c1d2)

## Sanity Checklist

- [ ] I have tested my changes using the available tooling
- [ ] I have commented my code
- [ ] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [ ] I have left TODOs throughout the codebase, if applicable

## Summary by CodeRabbit

- **Documentation**
  - Updated "DevNet" documentation to "DevNets" with new sections on interaction, types, manual provisioning, and testing.
  - Reorganized load testing documentation to clarify details and shifted focus from k6 to Gherkin.
  - Added comprehensive guides for load testing on DevNets and TestNets.
  - Adjusted sidebar position and table data for load testing plan documentation.
  
- **Chores**
  - Updated `.gitignore` to ignore temporary `job.yaml` files created by e2e scripts.

- **Refactor**
  - Updated `Makefile` to support custom manifest files for stress testing relays and renamed a target for clarity.

- **Configuration**
  - Updated `loadtest_manifest_example.yaml` and `loadtest_manifest_localnet.yaml` with new URLs, addresses, and comments for better clarity and alignment with configurations.

### CODE REVIEW

**Comment by okdas** [comment]
Hey @red-0ne - here are some updated load test documentation pages. Would you mind taking a look? Thank you!

**Comment by okdas** [comment]
@red-0ne actually had a chat with @bryanchriswhite - he'll take a look at this PR. You've got so many other PRs to review at this point! :)

File: docusaurus/docs/operate/testing/load_testing.md [hunk 1]
```diff
@@ -9,129 +9,100 @@ Poktroll load-testing suite.
...
-│   └── index.js
-├── modules                              # reusable code for scenarios and tests
-│   └── etheriumRequests.js
-├── scenarios                            # different scenarios for tests
-│   └── requestBlockNumberEtherium.js
-└── tests                                # test scripts
-    ├── anvilDirectEtherium.js
-    └── appGateServerEtherium.js
-```
+For custom test development, refer to the [gocuke documentation](https://github.com/regen-network/gocuke?tab=readme-ov-file#quick-start). The [anvil test](https://github.com/pokt-network/poktroll/blob/main/load-testing/tests/anvil_test.go) provides a practical reference implementation.
```
Comment (bryanchriswhite) [inline docusaurus/docs/operate/testing/load_testing.md:108]:
```suggestion
For custom test development, refer to the [gocuke documentation](https://github.com/regen-network/gocuke?tab=readme-ov-file#quick-start). The [anvil test](https://github.com/pokt-network/poktroll/blob/main/load-testing/tests/anvil_test.go) provides a small but practical reference implementation.
```

File: docusaurus/docs/operate/testing/load_testing_devnet.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+### 1. Create and configure the DevNet
+
+Please refer to the DevNet creation guide [here](../infrastructure/devnet.md#how-to-create).
+
+### 2. Stake the necessary actors
+
+- Depending on your load testing requirements, you may need to stake one or more `gateways` and `suppliers`.
+- [DevNet documentation](../infrastructure/devnet.md#stake-actors) provides more details about staking actors in DevNets.
+
+### 3. Configure the load testing manifest
```
Comment (bryanchriswhite) [inline docusaurus/docs/operate/testing/load_testing_devnet.md:37]:
```suggestion
### 3. Configure the load test manifest
```

File: docusaurus/docs/operate/testing/load_testing_testnet.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+Load Testing on TestNets is very similar to testing on DevNets. The main difference is that the test might run against software you don't control. Gateways and Suppliers can be hosted by other teams or organizations, so you need to ensure that your tests do not adversely affect their operations. As another side effect, you won't be able to collect metrics and logs on the software you don't run.
+
+## Load Testing Steps
+
+Please refer to the generic [load testing documentation](./load_testing.md#non-ephemeral-networks-testnets-mainnet-etc) for non-ephemeral networks (TestNets, MainNet, etc.). The steps are very similar to those for DevNets, which are thoroughly documented on the dedicated [Load Testing on DevNets](./load_testing_devnet.md) page.
+
+Key points to consider:
+1. Identify the TestNet endpoints and actors (gateways, suppliers) you'll be testing against.
+2. Ensure you have the necessary permissions and have communicated your intent to run load tests.
+3. Configure your load testing manifest to target the TestNet resources.
```
Comment (bryanchriswhite) [inline docusaurus/docs/operate/testing/load_testing_testnet.md:25]:
```suggestion
3. Configure your load test manifest to target the TestNet resources.
```

File: docusaurus/docs/operate/testing/load_testing_devnet.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+[Load Testing documentation](./load_testing.md#manifest-modification) provides information on how the load test manifest
+can be modified to run against DevNets. DevNets are not much different from TestNets, but they do not have randomly
+generated addresses. Instead, the addresses are shared with LocalNet for convenience.
```
Comment (bryanchriswhite) [inline docusaurus/docs/operate/testing/load_testing_devnet.md:41]:
```suggestion
can be modified to run against DevNets. DevNets are not much different from TestNets, but they **do not** have randomly
generated accounts. Instead, the accounts are derived from the ignite `config.yaml` (as in LocalNet) for convenience.
```

File: docusaurus/docs/operate/testing/load_testing_devnet.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+# testnet_node is the URL of the node that the load test will use to query the
+# chain and submit transactions.
+testnet_node: https://devnet-sophon-validator-rpc.poktroll.com
+
+# The service ID to request relays from.
+service_id: "anvil"
+
+# The address of the account that will be used to fund the application accounts
+# so that they can stake on the network.
+funding_account_address: pokt1eeeksh2tvkh7wzmfrljnhw4wrhs55lcuvmekkw # address for pnf account
```
Comment (bryanchriswhite) [inline docusaurus/docs/operate/testing/load_testing_devnet.md:92]:
```suggestion
# TODO_TECHDEBT(@bryanchriswhite, #512): Replace with faucet address.
funding_account_address: pokt1eeeksh2tvkh7wzmfrljnhw4wrhs55lcuvmekkw # address for pnf account
```

File: docusaurus/docs/operate/testing/load_testing_devnet.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+# In non-ephemeral chains, the gateways are identified by their address.
```
Comment (bryanchriswhite) [inline docusaurus/docs/operate/testing/load_testing_devnet.md:94]:
```suggestion
# In non-ephemeral chains, the gateways are identified by their addresses.
```

File: docusaurus/docs/operate/testing/load_testing_testnet.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+## Overview
+
+Load Testing on TestNets is very similar to testing on DevNets. The main difference is that the test might run against software you don't control. Gateways and Suppliers can be hosted by other teams or organizations, so you need to ensure that your tests do not adversely affect their operations. As another side effect, you won't be able to collect metrics and logs on the software you don't run.
+
+## Load Testing Steps
+
+Please refer to the generic [load testing documentation](./load_testing.md#non-ephemeral-networks-testnets-mainnet-etc) for non-ephemeral networks (TestNets, MainNet, etc.). The steps are very similar to those for DevNets, which are thoroughly documented on the dedicated [Load Testing on DevNets](./load_testing_devnet.md) page.
+
+Key points to consider:
```
Comment (bryanchriswhite) [inline docusaurus/docs/operate/testing/load_testing_testnet.md:22]: :ok_hand:

File: load-testing/loadtest_manifest_localnet.yaml
Comment (bryanchriswhite) [inline load-testing/loadtest_manifest_localnet.yaml:1]: :ok_hand:

**Review by bryanchriswhite — CHANGES_REQUESTED** [review]
My apologies for the delay on this review. :bowing_man: 

This is :fire::fire::fire: - thanks for pulling all of this together! :raised_hands: :rocket:

File: docusaurus/docs/operate/infrastructure/devnet.md [hunk 8]
```diff
@@ -54,3 +63,110 @@ Each DevNet ArgoCD App (following the App of Apps pattern) provisions a Helm cha
...
 Each Helm chart receives a list of configuration files. For example, see the [relayminer configuration](https://github.com/pokt-network/protocol-infra/blob/main/charts/full-network/templates/Application-Relayminer.yaml#L37). All possible values can be found in the `values.yaml` of the Helm chart, such as the [relayminer Helm chart](https://github.com/pokt-network/helm-charts/blob/main/charts/relayminer/values.yaml).
+
+## Interaction with DevNet
+
+:::note
+Devnets are provisioned with the same mnemonc phrases as LocalNet, so it is possible to reuse the same keys
+from the keybase - the user just need to change the `--node=` argument to point to the DevNet RPC endpoint.
+:::
+
+The DevNet RPC endpoint is exposed on `https://devnet-**NETWORK_NAME**-validator-rpc.poktroll.com`.
```
[the remark this answers is not in the document]
Comment (okdas) [reply docusaurus/docs/operate/infrastructure/devnet.md:74]: Would an example a few lines below count? 😅

**Comment by okdas** [comment]
Thank you, @bryanchriswhite. As always — amazing feedback! I addressed most of your points, and have a couple to take care of left. I'll request a review tomorrow once they're taken care of.

**Comment by okdas** [comment]
@bryanchriswhite all feedback has been addressed now. Thank you! :)

**Review by bryanchriswhite — APPROVED** [review]
:rocket: :rocket: :rocket: :rocket:

**Pull request merged**